include/oboe/FullDuplexStream.h

```diff
@@ -24,26 +24,68 @@
 
 namespace oboe {
 
-class FullDuplexStream : public AudioStreamCallback {
+/**
+ * FullDuplexStream can be used to synchronize an input and output stream.
+ *
+ * For the builder of the output stream, call setDataCallback() with this stream.
+ *
+ * When both streams are ready, onAudioReady() will call onBothStreamsReady().
+ * Callers must override onBothStreamsReady().
+ *
+ * To ensure best results, open an output stream before the input stream.
+ * Call inputBuilder.setBufferCapacityInFrames(mOutputStream->getBufferSizeInFrames() * 2).
+ *
+ * Callers must call setInputStream() and setOutputStream().
+ * Call start() to start both streams and stop() to stop both streams.
+ *
+ */
```

Comment:
What should we do about error callbacks? A disconnect on the output will close the output stream but not the input stream.  You can this in the "Echo Input to Output" test in OboeTester.

Reply:
I think the caller should restart both streams if either stream disconnects.

Reply:
Changed to the only subscribing to the error callback on the output stream.